Docs for next version (#6008)
apollographql/apollo-kotlin: docs/source/advanced/nullability.mdx

```

<Note>
The error is thrown during parsing but still caught before it reaches your UI code. If no parent field catches it, the Apollo Kotlin runtime does and exposes the exception in `ApolloResponse.exception`. 
</Note>  

## `@catchByDefault`

In order to use the nullability directives, you need to opt in a default catch behaviour for nullable GraphQL fields using `@catchByDefault`.

You can choose to map nullable fields to `FieldResult`:

```graphql
# Errors stop the parsing. 
extend schema @catchByDefault(to: RESULT)
```

Or throw errors:

```graphql
# Errors stop the parsing. 
extend schema @catchByDefault(to: THROW)
```

Or coerce errors to `null`, like the current GraphQL default:

```graphql
# Coerce errors to null by default.
extend schema @catchByDefault(to: NULL)
```

(Adding `@catchByDefault(to: NULL)` is a no-op for codegen that unlocks using `@catch` in your operations.)

## Migrate to semantic nullability

Semantic nullability is the most useful for schemas that are nullable by default. These are the schemas that require "handling every field error". 

In order to change that default to "opt-in the errors you want to handle", you can use the following approach:

> 1. import the nullability directives.

Reviewer: ```suggestion
1. Import the nullability directives.
```

1. Default to coercing to null: `extend schema @catchByDefault(to: NULL)`. This is a no-op to start exploring the directives.
1. Add `@catch` to individual fields, get more comfortable with how it works.
1. When ready to do the big switch, change to `extend schema catch(to: THROW)` and (at the same time) add  `query GetFoo @catch(to: NULL) {}` on all operations/fragments (this is a no-op).

> 1. From this moment on, new queries written are `catch(to: THROW)` by default.

Reviewer: I find this sentence not 100% clear - maybe something like:

```suggestion
1. From this moment on, new queries will be `catch(to: THROW)` by default.
```
?
